libraries/AP_NavEKF3/AP_NavEKF3_core.cpp

```diff
@@ -1982,6 +1982,17 @@ void NavEKF3_core::MagTableConstrain(void)
                                                    table_earth_field_ga.z+limit_ga);
 }
 
+// constrain states using limit on mag offsets
+void NavEKF3_core::MagOffsetConstrain(void)
+{
+    if (frontend->_mag_learn_limit > 0) {
+        const float limit_ga = frontend->_mag_learn_limit * 0.001;
+        stateStruct.body_magfield.x = constrain_float(stateStruct.body_magfield.x, -limit_ga, limit_ga);
```

Comment:
So these limits are applied on a per-axis basis.  I guess you've considered instead limiting the overall length (e.g. scaling instead of constraining)?

Reply:
yes, its deliberate, and matches the existing EK3_MAG_EF_LIM